Dockerfile

```diff
@@ -1,19 +1,43 @@
-FROM node:22-slim AS base
-ENV PNPM_HOME="/pnpm"
-ENV PATH="$PNPM_HOME:$PATH"
-RUN corepack enable
+# install and optimize dependencies
+FROM zenika/alpine-chrome:with-node AS base
+
 WORKDIR /app
+
+# copy package files
 COPY package.json pnpm-lock.yaml ./
-RUN corepack prepare
 
-FROM base AS prod-deps
-RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --prod --frozen-lockfile
+# install pnpm as root,
+# then switch back to chrome user
+USER root
+RUN npm install -g pnpm@10
+
+# install dependencies as normal user
+USER chrome
+RUN pnpm install --prod --frozen-lockfile && \
+    pnpm prune --prod && \
+    # remove unnecessary files to reduce image size
+    find /app/node_modules -name "*.md" -delete && \
+    find /app/node_modules -type d -name "test*" -exec rm -rf {} + 2>/dev/null || true && \
+    find /app/node_modules -type d -name "doc*" -exec rm -rf {} + 2>/dev/null || true && \
+    find /app/node_modules -name ".cache" -type d -exec rm -rf {} + 2>/dev/null || true && \
+    # clean up caches
+    rm -rf ~/.pnpm ~/.npm /tmp/* /var/cache/apk/*
```

Comment:
Instead of this, we can try and use tool we use in Open Runtimes; it knows a bit more files and paths:

https://github.com/open-runtimes/open-runtimes/blob/main/runtimes/node/versions/latest/helpers/sveltekit/bundle.sh#L17